x/ccv/provider/keeper/keeper.go

```diff
@@ -1136,3 +1136,51 @@ func (k Keeper) GetAllRegisteredAndProposedChainIDs(ctx sdk.Context) []string {
 
 	return allConsumerChains
 }
+
+// SetTopN stores the N value associated to chain with `chainID`
+func (k Keeper) SetTopN(
+	ctx sdk.Context,
+	chainID string,
+	N uint32,
+) {
+	store := ctx.KVStore(k.storeKey)
+
+	buf := make([]byte, 4)
+	binary.BigEndian.PutUint32(buf, N)
+
+	store.Set(types.TopNKey(chainID), buf)
+}
+
+// DeleteTopN removes the N value associated to chain with `chainID`
+func (k Keeper) DeleteTopN(
+	ctx sdk.Context,
+	chainID string,
+) {
+	store := ctx.KVStore(k.storeKey)
+	store.Delete(types.TopNKey(chainID))
+}
+
+// GetTopN returns a pair the N associated to chain with `chainID`
+func (k Keeper) GetTopN(
+	ctx sdk.Context,
+	chainID string,
+) (uint32, bool) {
+	store := ctx.KVStore(k.storeKey)
+	buf := store.Get(types.TopNKey(chainID))
+	if buf == nil {
+		return 0, false
+	}
+	return binary.BigEndian.Uint32(buf), true
+}
+
+// IsTopN returns true if chain with `chainID` is a Top N chain (i.e., enforces at least one validator to validate chain `chainID`)
+func (k Keeper) IsTopN(ctx sdk.Context, chainID string) bool {
+	topN, found := k.GetTopN(ctx, chainID)
+	return found && topN > 0
+}
+
+// IsOptIn returns true if chain with `chainID` is an Opt In chain (i.e., no validator is forced to validate chain `chainID`)
+func (k Keeper) IsOptIn(ctx sdk.Context, chainID string) bool {
```

Comment:
This is strange, indeed. The problem stems from the fact that we use `uint32` which has a zero value of `0`. It would be impossible to distinguish on whether a user that submits a `ConsumerAdditionProposal` sets `0`  to `Top_N` or does not include `Top_N` at all. Because, if a user  forgets to set `Top_N`, it would be considered the same as `0` when we look at it.
I guess a solution would be to use a `string` or something and if the `string` is empty, then we can infer  that the user did not provide a value, but I'm not sure if it's worth it.

At the end of the day, everything has to go through a proposal, so a proposal would be scrutinized before it passes. If for a proposal we have `Top_N = 0` but the proposal is for a Top N chain, then something obviously went wrong with this proposal and it should be rejected.